packages/clerk-js/src/ui/primitives/Image.tsx

```diff
@@ -1,14 +1,23 @@
 import React from 'react';
 
+import { generateSrc, generateSrcSet } from '../../utils';
 import { PrimitiveProps, StateProps } from '../styledSystem';
 import { applyDataStateProps } from './applyDataStateProps';
 
-export type ImageProps = PrimitiveProps<'img'> & StateProps;
+export type ImageProps = PrimitiveProps<'img'> &
+  StateProps & {
+    size?: number;
+    xDescriptors?: number[];
+  };
 
 export const Image = React.forwardRef<HTMLImageElement, ImageProps>((props, ref) => {
+  const { src, size = 80, xDescriptors = [1, 2], ...rest } = props;
+
   return (
     <img
-      {...applyDataStateProps(props)}
+      srcSet={generateSrcSet({ src, width: size, xDescriptors })}
+      src={generateSrc({ src, width: size * 2 })}
```

Comment:
❓ Do we always want to fetch an image twice the size of the img element by default?

Reply:
If `srcset` is not supported, then we would want to support the retina displays. Is that not reason enough?